Docker build fails

### Issue Description

When trying to build the docker image, the build fails.

### Steps to Reproduce

Open new Gitpod Workspace and run:
`docker compose -f compose-test.yml  up --build`

### Expected Behavior

_No response_

### Current Behavior

```bash
$ docker compose -f compose-test.yml  up --build
[+] Building 76.3s (23/31)                                                                                                                                                      docker:default
 => [documenso internal] load .dockerignore                                                                                                                                               0.0s
 => => transferring context: 429B                                                                                                                                                         0.0s
 => [documenso internal] load build definition from Dockerfile                                                                                                                            0.0s
 => => transferring dockerfile: 2.82kB                                                                                                                                                    0.0s
 => [documenso internal] load metadata for docker.io/library/node:18-alpine                                                                                                               0.6s
 => [documenso internal] load build context                                                                                                                                               0.3s
 => => transferring context: 1.37MB                                                                                                                                                       0.2s
 => [documenso base 1/1] FROM docker.io/library/node:18-alpine@sha256:3428c2de886bf4378657da6fe86e105573a609c94df1f7d6a70e57d2b51de21f                                                    0.0s
 => CACHED [documenso builder 1/7] RUN apk add --no-cache libc6-compat                                                                                                                    0.0s
 => CACHED [documenso builder 2/7] RUN apk add --no-cache jq                                                                                                                              0.0s
 => CACHED [documenso builder 3/7] WORKDIR /app                                                                                                                                           0.0s
 => [documenso builder 4/7] COPY . .                                                                                                                                                      3.2s
 => [documenso builder 5/7] RUN TURBO_VERSION="$(npm list --package-lock-only --json turbo | jq -r '.dependencies.turbo.version')"                                                        2.1s
 => [documenso builder 6/7] RUN npm install -g "turbo@$TURBO_VERSION"                                                                                                                     2.3s
 => [documenso builder 7/7] RUN turbo prune --scope=@documenso/web --docker                                                                                                               0.7s
 => CACHED [documenso installer  3/13] RUN apk add --no-cache make cmake g++                                                                                                              0.0s 
 => CACHED [documenso installer  4/13] WORKDIR /app                                                                                                                                       0.0s 
 => CACHED [documenso installer  5/13] COPY .gitignore .gitignore                                                                                                                         0.0s 
 => [documenso installer  6/13] COPY --from=builder /app/out/json/ .                                                                                                                      0.0s 
 => [documenso installer  7/13] COPY --from=builder /app/out/package-lock.json ./package-lock.json                                                                                        0.0s 
 => [documenso installer  8/13] RUN npm ci                                                                                                                                               24.6s
 => [documenso installer  9/13] COPY --from=builder /app/out/full/ .                                                                                                                      0.2s
 => [documenso installer 10/13] COPY turbo.json turbo.json                                                                                                                                0.0s 
 => [documenso installer 11/13] RUN TURBO_VERSION="$(npm list --package-lock-only --json turbo | jq -r '.dependencies.turbo.version')"                                                    1.8s
 => [documenso installer 12/13] RUN npm install -g "turbo@$TURBO_VERSION"                                                                                                                 2.2s
 => ERROR [documenso installer 13/13] RUN turbo run build --filter=@documenso/web...                                                                                                     36.8s
------
 > [documenso installer 13/13] RUN turbo run build --filter=@documenso/web...:
0.834 • Packages in scope: @documenso/ee, @documenso/email, @documenso/lib, @documenso/prisma, @documenso/signing, @documenso/tailwind-config, @documenso/trpc, @documenso/tsconfig, @documenso/ui, @documenso/web
0.834 • Running build in 10 packages
0.834 • Remote caching disabled
0.835 @documenso/prisma:build: cache miss, executing e4677164279358f4
1.194 @documenso/prisma:build: 
1.194 @documenso/prisma:build: > @documenso/prisma@1.0.0 build
1.194 @documenso/prisma:build: > prisma generate
1.194 @documenso/prisma:build: 
2.050 @documenso/prisma:build: Prisma schema loaded from schema.prisma
2.583 @documenso/prisma:build: 
2.583 @documenso/prisma:build: ✔ Generated Prisma Client (v5.4.2) to ./../../node_modules/@prisma/client in 204ms
2.583 @documenso/prisma:build: 
2.583 @documenso/prisma:build: Start using Prisma Client in Node.js (See: https://pris.ly/d/client)
2.583 @documenso/prisma:build: ```
2.583 @documenso/prisma:build: import { PrismaClient } from '@prisma/client'
2.583 @documenso/prisma:build: const prisma = new PrismaClient()
2.583 @documenso/prisma:build: ```
2.583 @documenso/prisma:build: or start using Prisma Client at the edge (See: https://pris.ly/d/accelerate)
2.583 @documenso/prisma:build: ```
2.583 @documenso/prisma:build: import { PrismaClient } from '@prisma/client/edge'
2.583 @documenso/prisma:build: const prisma = new PrismaClient()
2.583 @documenso/prisma:build: ```
2.583 @documenso/prisma:build: 
2.583 @documenso/prisma:build: See other ways of importing Prisma Client: http://pris.ly/d/importing-client
2.583 @documenso/prisma:build: 
2.623 @documenso/web:build: cache miss, executing d95e4354cefe1c9f
2.968 @documenso/web:build: 
2.968 @documenso/web:build: > @documenso/web@0.1.0 build
2.968 @documenso/web:build: > next build
2.968 @documenso/web:build: 
3.694 @documenso/web:build:    ▲ Next.js 14.0.0
3.694 @documenso/web:build:    - Experiments (use at your own risk):
3.694 @documenso/web:build:      · outputFileTracingRoot
3.695 @documenso/web:build: 
3.695 @documenso/web:build:    Creating an optimized production build ...
3.716 @documenso/web:build:  ⚠ Custom webpack configuration is detected. When using a custom webpack configuration, the Webpack build worker is disabled by default. To force enable it, set the "experimental.webpackBuildWorker" option to "true". Read more: https://nextjs.org/docs/messages/webpack-build-worker-opt-out
35.95 @documenso/web:build: <w> [webpack.cache.PackFileCacheStrategy] Serializing big strings (126kiB) impacts deserialization performance (consider using Buffer instead and decode when needed)
36.49 @documenso/web:build: Failed to compile.
36.49 @documenso/web:build: 
36.49 @documenso/web:build: ./src/components/partials/not-found.tsx
36.49 @documenso/web:build: Module not found: Can't resolve '@documenso/assets/images/background-pattern.png'
36.49 @documenso/web:build: 
36.49 @documenso/web:build: https://nextjs.org/docs/messages/module-not-found
36.49 @documenso/web:build: 
36.49 @documenso/web:build: ./src/app/(signing)/sign/[token]/complete/page.tsx
36.49 @documenso/web:build: Module not found: Can't resolve '@documenso/assets/images/signing-celebration.png'
36.49 @documenso/web:build: 
36.49 @documenso/web:build: https://nextjs.org/docs/messages/module-not-found
36.49 @documenso/web:build: 
36.49 @documenso/web:build: ./src/app/(unauthenticated)/layout.tsx
36.49 @documenso/web:build: Module not found: Can't resolve '@documenso/assets/images/background-pattern.png'
36.49 @documenso/web:build: 
36.49 @documenso/web:build: https://nextjs.org/docs/messages/module-not-found
36.49 @documenso/web:build: 
36.49 @documenso/web:build: ../../packages/ee/server-only/stripe/webhook/on-early-adopters-checkout.ts
36.49 @documenso/web:build: Module not found: Can't resolve '@documenso/assets/documenso-supporter-pledge.pdf'
36.49 @documenso/web:build: 
36.49 @documenso/web:build: https://nextjs.org/docs/messages/module-not-found
36.49 @documenso/web:build: 
36.49 @documenso/web:build: Import trace for requested module:
36.49 @documenso/web:build: ../../packages/ee/server-only/stripe/webhook/handler.ts
36.49 @documenso/web:build: ./src/pages/api/stripe/webhook/index.ts
36.49 @documenso/web:build: 
36.49 @documenso/web:build: ./src/app/(share)/share/[slug]/opengraph/route.tsx:26:10
36.49 @documenso/web:build: Module not found: Can't resolve '@documenso/assets/fonts/inter-semibold.ttf'
36.49 @documenso/web:build:   24 | export async function GET(_request: Request, { params: { slug } }: SharePageOpenGraphImageProps) {
36.49 @documenso/web:build:   25 |   const [interSemiBold, interRegular, caveatRegular, shareFrameImage] = await Promise.all([
36.49 @documenso/web:build: > 26 |     fetch(new URL('@documenso/assets/fonts/inter-semibold.ttf', import.meta.url)).then(
36.49 @documenso/web:build:      |          ^
36.49 @documenso/web:build:   27 |       async (res) => res.arrayBuffer(),
36.49 @documenso/web:build:   28 |     ),
36.49 @documenso/web:build:   29 |     fetch(new URL('@documenso/assets/fonts/inter-regular.ttf', import.meta.url)).then(async (res) =>
36.49 @documenso/web:build: 
36.49 @documenso/web:build: https://nextjs.org/docs/messages/module-not-found
36.49 @documenso/web:build: 
36.49 @documenso/web:build: Import trace for requested module:
36.49 @documenso/web:build: ../../node_modules/next/dist/build/webpack/loaders/next-edge-app-route-loader/index.js?absolutePagePath=private-next-app-dir%2F(share)%2Fshare%2F%5Bslug%5D%2Fopengraph%2Froute.tsx&page=%2F(share)%2Fshare%2F%5Bslug%5D%2Fopengraph%2Froute&appDirLoader=bmV4dC1hcHAtbG9hZGVyP25hbWU9YXBwJTJGKHNoYXJlKSUyRnNoYXJlJTJGJTVCc2x1ZyU1RCUyRm9wZW5ncmFwaCUyRnJvdXRlJnBhZ2U9JTJGKHNoYXJlKSUyRnNoYXJlJTJGJTVCc2x1ZyU1RCUyRm9wZW5ncmFwaCUyRnJvdXRlJnBhZ2VQYXRoPXByaXZhdGUtbmV4dC1hcHAtZGlyJTJGKHNoYXJlKSUyRnNoYXJlJTJGJTVCc2x1ZyU1RCUyRm9wZW5ncmFwaCUyRnJvdXRlLnRzeCZhcHBEaXI9JTJGYXBwJTJGYXBwcyUyRndlYiUyRnNyYyUyRmFwcCZhcHBQYXRocz0lMkYoc2hhcmUpJTJGc2hhcmUlMkYlNUJzbHVnJTVEJTJGb3BlbmdyYXBoJTJGcm91dGUmcGFnZUV4dGVuc2lvbnM9dHN4JnBhZ2VFeHRlbnNpb25zPXRzJnBhZ2VFeHRlbnNpb25zPWpzeCZwYWdlRXh0ZW5zaW9ucz1qcyZiYXNlUGF0aD0mYXNzZXRQcmVmaXg9Jm5leHRDb25maWdPdXRwdXQ9c3RhbmRhbG9uZSZwcmVmZXJyZWRSZWdpb249Jm1pZGRsZXdhcmVDb25maWc9ZTMwJTNEIQ%3D%3D&nextConfigOutput=standalone&preferredRegion=&middlewareConfig=e30%3D!
36.49 @documenso/web:build: 
36.49 @documenso/web:build: 
36.49 @documenso/web:build: > Build failed because of webpack errors
36.56 @documenso/web:build: npm ERR! Lifecycle script `build` failed with error: 
36.57 @documenso/web:build: npm ERR! Error: command failed 
36.57 @documenso/web:build: npm ERR!   in workspace: @documenso/web@0.1.0 
36.57 @documenso/web:build: npm ERR!   at location: /app/apps/web 
36.58 @documenso/web:build: ERROR: command finished with error: command (/app/apps/web) npm run build exited (1)
36.58 @documenso/web#build: command (/app/apps/web) npm run build exited (1)
36.58 
36.58  Tasks:    1 successful, 2 total
36.58 Cached:    0 cached, 2 total
36.58   Time:    35.978s 
36.58 Failed:    @documenso/web#build
36.58 
36.58  ERROR  run failed: command  exited (1)
------
failed to solve: process "/bin/sh -c turbo run build --filter=@documenso/web..." did not complete successfully: exit code: 1
```

### Screenshots (optional)

_No response_

### Operating System [e.g., Windows 10]

Gitpod

### Browser [e.g., Chrome, Firefox]

_No response_

### Version [e.g., 2.0.1]

main (fbbc3b89c3eceef2129b03a86ed66be7b316a32a)

### Please check the boxes that apply to this issue report.

- [X] I have searched the existing issues to make sure this is not a duplicate.
- [X] I have provided steps to reproduce the issue.
- [X] I have included relevant environment information.
- [X] I have included any relevant screenshots.
- [X] I understand that this is a voluntary contribution and that there is no guarantee of resolution.
- [ ] I want to work on creating a PR for this issue if approved

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -14,8 +14,8 @@ env:
   HUSKY: 0
 
 jobs:
-  build:
-    name: Build
+  build_app:
+    name: Build App
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
@@ -37,8 +37,9 @@ jobs:
 
       - name: Build
         run: npm run build
+
   build_docker:
-    name: Build
+    name: Build Docker Image
     runs-on: ubuntu-latest
     steps:
       - name: Checkout

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -37,3 +37,15 @@ jobs:
 
       - name: Build
         run: npm run build
+  build_docker:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Build Docker Image
+        run: ./docker/build.sh
+

--- a/apps/marketing/package.json
+++ b/apps/marketing/package.json
@@ -13,6 +13,7 @@
     "copy:pdfjs": "node ../../scripts/copy-pdfjs.cjs"
   },
   "dependencies": {
+    "@documenso/assets": "*",
     "@documenso/lib": "*",
     "@documenso/tailwind-config": "*",
     "@documenso/trpc": "*",

--- a/apps/web/package.json
+++ b/apps/web/package.json
@@ -13,6 +13,7 @@
     "copy:pdfjs": "node ../../scripts/copy-pdfjs.cjs"
   },
   "dependencies": {
+    "@documenso/assets": "*",
     "@documenso/ee": "*",
     "@documenso/lib": "*",
     "@documenso/prisma": "*",

--- a/apps/web/src/components/(dashboard)/layout/desktop-nav.tsx
+++ b/apps/web/src/components/(dashboard)/layout/desktop-nav.tsx
@@ -16,7 +16,9 @@ export const DesktopNav = ({ className, ...props }: DesktopNavProps) => {
   // const pathname = usePathname();
   const [open, setOpen] = useState(false);
 
-  const isMacOS = /Macintosh|Mac\s+OS\s+X/i.test(navigator?.userAgent || 'unknown');
+  const userAgent = typeof navigator !== 'undefined' ? navigator.userAgent : 'unknown';
+
+  const isMacOS = /Macintosh|Mac\s+OS\s+X/i.test(userAgent);
   const modifierKey = isMacOS ? '⌘' : 'Ctrl';
 
   return (

--- a/packages/email/mailer.ts
+++ b/packages/email/mailer.ts
@@ -42,12 +42,8 @@ const getTransport = () => {
     });
   }
 
-  if (!process.env.NEXT_PRIVATE_SMTP_HOST) {
-    throw new Error('SMTP transport requires NEXT_PRIVATE_SMTP_HOST');
-  }
-
   return createTransport({
-    host: process.env.NEXT_PRIVATE_SMTP_HOST,
+    host: process.env.NEXT_PRIVATE_SMTP_HOST ?? 'localhost:2500',
     port: Number(process.env.NEXT_PRIVATE_SMTP_PORT) || 587,
     secure: process.env.NEXT_PRIVATE_SMTP_SECURE === 'true',
     auth: {

